### PR TITLE
don't crash during container removal if the container doesn't exist

### LINKED ISSUE #95: Cannot start containers anymore after update
After clicking update in the AIO interface to get to the latest version and trying to restart the containers via tha UI, they don't start anymore at all, the XHR request done in the background throws the following error:

```
<b>Fatal error</b>:  Uncaught GuzzleHttp\Exception\ClientException: Client error: `POST http://localhost/v1.41/networks/nextcloud-aio/disconnect` resulted in a `404 Not Found` response:
{&quot;message&quot;:&quot;No such container: nextcloud-aio-domaincheck&quot;}

 in /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113
Stack trace:
#0 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Middleware.php(69): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response), NULL, Array, NULL)
#1 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(204): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response))
#2 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(153): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), NULL)
#3 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
#4 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Promise\TaskQueue-&gt;run(true)
#5 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise-&gt;invokeWaitFn()
#6 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise-&gt;waitIfPending()
#7 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise-&gt;invokeWaitList()
#8 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\Promise\Promise-&gt;waitIfPending()
#9 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Client.php(187): GuzzleHttp\Promise\Promise-&gt;wait()
#10 /var/www/docker-aio/php/src/Docker/DockerActionManager.php(425): GuzzleHttp\Client-&gt;request('POST', 'http://localhos...', Array)
#11 /var/www/docker-aio/php/src/Controller/DockerController.php(135): AIO\Docker\DockerActionManager-&gt;DisconnectContainerFromNetwork(Object(AIO\Container\Container))
#12 /var/www/docker-aio/php/src/Controller/DockerController.php(171): AIO\Controller\DockerController-&gt;PerformRecursiveContainerStop('nextcloud-aio-d...')
#13 /var/www/docker-aio/php/src/Controller/DockerController.php(113): AIO\Controller\DockerController-&gt;StopDomaincheckContainer()
#14 /var/www/docker-aio/php/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(43): AIO\Controller\DockerController-&gt;StartContainer(Object(GuzzleHttp\Psr7\ServerRequest), Object(GuzzleHttp\Psr7\Response), Array)
#15 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(384): Slim\Handlers\Strategies\RequestResponse-&gt;__invoke(Array, Object(GuzzleHttp\Psr7\ServerRequest), Object(GuzzleHttp\Psr7\Response), Array)
#16 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Slim\Routing\Route-&gt;handle(Object(GuzzleHttp\Psr7\ServerRequest))
#17 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Slim\MiddlewareDispatcher-&gt;handle(Object(GuzzleHttp\Psr7\ServerRequest))
#18 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(341): Slim\MiddlewareDispatcher-&gt;handle(Object(GuzzleHttp\Psr7\ServerRequest))
#19 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/RouteRunner.php(84): Slim\Routing\Route-&gt;run(Object(GuzzleHttp\Psr7\ServerRequest))
#20 /var/www/docker-aio/php/vendor/slim/csrf/src/Guard.php(456): Slim\Routing\RouteRunner-&gt;handle(Object(GuzzleHttp\Psr7\ServerRequest))
#21 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(209): Slim\Csrf\Guard-&gt;process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Slim\Routing\RouteRunner))
#22 /var/www/docker-aio/php/vendor/slim/twig-view/src/TwigMiddleware.php(125): Psr\Http\Server\RequestHandlerInterface@anonymous-&gt;handle(Object(GuzzleHttp\Psr7\ServerRequest))
#23 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(147): Slim\Views\TwigMiddleware-&gt;process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#24 /var/www/docker-aio/php/src/Middleware/AuthMiddleware.php(38): Psr\Http\Server\RequestHandlerInterface@anonymous-&gt;handle(Object(GuzzleHttp\Psr7\ServerRequest))
#25 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(313): AIO\Middleware\AuthMiddleware-&gt;__invoke(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#26 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Psr\Http\Server\RequestHandlerInterface@anonymous-&gt;handle(Object(GuzzleHttp\Psr7\ServerRequest))
#27 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(215): Slim\MiddlewareDispatcher-&gt;handle(Object(GuzzleHttp\Psr7\ServerRequest))
#28 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(199): Slim\App-&gt;handle(Object(GuzzleHttp\Psr7\ServerRequest))
#29 /var/www/docker-aio/php/public/index.php(140): Slim\App-&gt;run()
#30 {main}
  thrown in <b>/var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php</b> on line <b>113</b><br />
```

The mentioned container indeed does not exist. I also tried troubleshooting by removing all containers (leaving their volumes in place) and running the bootstrap command from `README.md` again - with the same result. It seems that `nextcloud-aio-domaincheck` is not created in the first place.

I cannot even manually restart the containers from the CLI as the nextcloud container seems to infinitely wait for the database container, although it's up. I don't know how the link is set up by the master container when spawning the others, but it seems like nextcloud and database don't have any connectivity anymore. I installed ping in the nc container and tried to ping the database container but even the DNS lookup fails. I didn't do anything on the system for weeks, so I don't assume it's an environment problem.

### PR DESCRIPTION
**Pull request by szaimen**
This is best reviewed like this: https://github.com/nextcloud/all-in-one/pull/96/files?diff=unified&w=1

Fix #95

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

File: php/src/Docker/DockerActionManager.php [hunk 1]
```diff
@@ -125,7 +125,11 @@ public function DeleteContainer(Container $container) {
         $url = $this->BuildApiUrl(sprintf('containers/%s', urlencode($container->GetIdentifier())));
         try {
             $this->guzzleClient->delete($url);
-        } catch (\Exception $e) {}
+        } catch (ClientException $e) {
```
Comment (juliushaertl) [inline php/src/Docker/DockerActionManager.php:128]:
```suggestion
        } catch (\GuzzleHttp\Exception\RequestException $e) {
```

https://docs.guzzlephp.org/en/stable/quickstart.html#exceptions

File: php/src/Docker/DockerActionManager.php [hunk 2]
```diff
@@ -426,7 +430,7 @@ public function DisconnectContainerFromNetwork(Container $container)
                     ],
                 ]
             );
-        } catch (ServerException $e) {}
+        } catch (ClientException $e) {}
```
Comment (juliushaertl) [inline php/src/Docker/DockerActionManager.php:433]:
```suggestion
        } catch (\GuzzleHttp\Exception\RequestException $e) {
```

File: php/src/Docker/DockerActionManager.php [hunk 3]
```diff
@@ -482,7 +486,9 @@ public function ConnectContainerToNetwork(Container $container)
 
     public function StopContainer(Container $container) {
         $url = $this->BuildApiUrl(sprintf('containers/%s/stop?t=%s', urlencode($container->GetIdentifier()), $container->GetMaxShutdownTime()));
-        $this->guzzleClient->post($url);
+        try {
+            $this->guzzleClient->post($url);
+        } catch (\Exception $e) {}
```
Comment (juliushaertl) [inline php/src/Docker/DockerActionManager.php:491]: Maybe have the same logic here to still throw the exception if the status code is not a 404?

**Review by juliushaertl — CHANGES_REQUESTED** [review]
See inline :wink:

File: php/src/Docker/DockerActionManager.php (hunk 1 above)
Comment (szaimen) [reply php/src/Docker/DockerActionManager.php]: done, thanks!

File: php/src/Docker/DockerActionManager.php (hunk 2 above)
Comment (szaimen) [reply php/src/Docker/DockerActionManager.php]: done

File: php/src/Docker/DockerActionManager.php (hunk 3 above)
Comment (szaimen) [reply php/src/Docker/DockerActionManager.php]: should be done with https://github.com/nextcloud/all-in-one/pull/96/commits/655eada45c46da7fb2e886afa682f9979bee2861

**Comment by szaimen** [comment]
Should be done now. Please have another look :)

**Review by juliushaertl — APPROVED** [review]
Code makes sense

**Pull request merged**